File: C Sharp/C Sharp/QuickSort.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace C_Sharp
{
    public static class QuickSort
    {
        private static int Partition(int[] array, int low, int high)
        {
            int x = array[high];
            int i = low - 1;
            int swap;

            for (int j = low; j <= high - 1; j++)
            {
                if (array[j] <= x)
                {
                    i++;
                    swap = array[j];
                    array[j] = array[i];
                    array[i] = swap;
                }
            }
            swap = array[i + 1];
            array[i + 1] = array[high];
            array[high] = swap;


            return i + 1;
        }

        public static void Sort(int[] array, int low, int high)
        {
            if (low < high)
            {
                int part = Partition(array, low, high);
                Sort(array, low, part - 1);
                Sort(array, part + 1, high);
            }
        }
    }
}

File: Python/QuickSort.py
class QuickSort:

    @staticmethod
    def __partition(array: list, low: int, high: int) -> int:
        x = array[high]
        i = low - 1

        for j in range(low, high, 1):
            if array[j] <= x:
                i += 1
                array[i], array[j] = array[j], array[i]

        array[i + 1], array[high] = array[high], array[i + 1]
        return i + 1

    def sort(self, array: list, low: int, high: int):
        if low < high:
            part = self.__partition(array, low, high)
            # print(array)
            self.sort(array, low, part - 1)
            self.sort(array, part + 1, high)


if __name__ == '__main__':
    arr = [2, 8, 7, 1, 3, 5, 6, 4]
    qs = QuickSort()
    qs.sort(arr, 0, len(arr) - 1)

    print(arr)
